tools/integration_tests/util/client/storage_client.go

```diff
@@ -222,3 +223,59 @@ func StatObject(ctx context.Context, client *storage.Client, object string) (*st
 	}
 	return attrs, nil
 }
+
+func UploadGcsObject(localPath, bucketName, objectName string, uploadGzipEncoded bool) error {
+	ctx := context.Background()
```

Comment:
Pass context into this function for better control, rather than using context.Background here.